frontend/src/components/withdrawal/WithdrawalButton.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="withdraw-button">
+    <div :class="withdrawingClass(isSending)" >
+      <green-button
+        class="wb-green-button"
+        @click.native='transferWithdrawal'
+        :text="'WITHDRAW ' + tokensGroupedByName + ' '"
+      />
+      <div class="wb-spinner-container">
+        <spinner />
+        <div class="wb-sending-text">
+          Please don't close this window and wait until we process your order. It usually takes about 10 seconds.
+        </div>
+      </div>
+      <div v-if="showErrorMsg" class="wb-error-occured">An error occured, please try again.</div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import GreenButton from 'components/common/GreenButton.vue';
+import Spinner from 'components/common/Spinner.vue';
+
+import { StateT } from 'store';
+import { GET_MY_BALANCE_ACTION } from 'store/account/actions';
+
+import { GET_MY_LOANS_ACTION } from 'store/my-loans/actions';
+import { GET_MY_INVESTMENTS_ACTION } from 'store/invest/actions';
+
+import { formatBigNumber } from 'utils/calc';
+import { withdrawAll } from 'api/withdrawal';
+
+export default {
+  name: 'MyLoanTilePaybackSection',
+  props: ['loanShortId', 'withdrawals'],
+  components: {
+    'green-button': GreenButton,
+    'spinner': Spinner,
+  },
+  data() {
+    return {
+      isSending: false,
+      showErrorMsg: false
+    };
+  },
+  computed: {
+    tokensGroupedByName: function() {
+      const tokenSymbolToAmountSum = this.withdrawals.reduce((acc, withdrawal) => {
+        const { amount, tokenSymbol } = withdrawal;
+        if (tokenSymbol in acc) {
+          acc[tokenSymbol] = acc[tokenSymbol].add(amount);
+        } else {
+          acc[tokenSymbol] = withdrawal.amount;
+        }
+        return acc;
+      }, {});
+
+      const withdrawalsSummaryPerToken = Object.keys(tokenSymbolToAmountSum).map(tokenSymbol => {
+        return {
+          tokenSymbol,
+          amount: tokenSymbolToAmountSum[tokenSymbol]
+        }
+      });
+
+      function format(withdrawal) {
+        return formatBigNumber(withdrawal.amount) + " " + withdrawal.tokenSymbol;
+      }
+
+      let buttonText = format(withdrawalsSummaryPerToken[0]);
+      for (let i = 1; i < withdrawalsSummaryPerToken.length; ++i) {
```

Comment:
`for (const element of withdrawalsSummaryPerToken) {`

Reply:
`withdrawalsSummaryPerToken.forEach()` is also pretty cool